src/ff1.rs

```diff
@@ -127,50 +127,46 @@ impl Radix {
     }
 }
 
-/// An integer.
-pub trait Numeral {
+/// Type representing FF1 operations that can be performed on a sub-section of a
+/// [`NumeralString`].
+pub trait Operations: Sized {
     /// Type used for byte representations.
     type Bytes: AsRef<[u8]>;
 
-    /// Returns the integer interpreted from the given bytes in big-endian order.
-    fn from_bytes(s: impl Iterator<Item = u8>) -> Self;
+    /// Returns the number of numerals in this numeral sub-string.
+    fn numeral_count(&self) -> usize;
 
-    /// Returns the big-endian byte representation of this integer.
-    fn to_bytes(&self, b: usize) -> Self::Bytes;
+    /// Encodes the number that this numeral string represents in the base
+    /// radix as a byte string, by valuing the numerals in decreasing order
+    /// of significance (big-endian order).
+    fn to_be_bytes(&self, radix: u32, b: usize) -> Self::Bytes;
 
-    /// Compute (self + other) mod radix^m
-    fn add_mod_exp(self, other: Self, radix: u32, m: usize) -> Self;
+    /// Compute (self + other) mod radix^m, where other is big-endian.
```

Comment:
```suggestion
    /// Compute (self + other) mod radix^m, where other is represented
    /// as a big-endian byte sequence.
```

Reply:
Saying big-endian here isn't helpful, I think. `self` and `other` both conceptually just represent integers at this level.